### DISCUSSION TITLE
Show legend with just one entity

### DISCUSSION THREAD

**Discussion by Digitalfm** [RomRider/apexcharts-card#798] [General]
Hi, would be really grateful if someone could tell me the YAML command to show the legend when the graph contains only one entity?

Currently the legend only appears when more than one entity is shown in the graph.

**Comment by sit-together** [comment]
Hello together,

I wrote the following configuration with embedded Javascript for my gas consumption with ISO calender weeks on the x-axis.
Works so far, with one exception: the last value is only shown if I skip the "Math.floor" in the line before the return line - hmmmm.

I add two pictures of the results - the first WITH Math.floor and the second WITHOUT Math.floor. As you see, the first doesn't show week 43, the second does ...

Thanks in advance for help!

<img width="496" alt="with_math_floor" src="https://github.com/user-attachments/assets/09c409f3-2d88-4c1d-9125-f2ee714b6218">

<img width="496" alt="without_math_floor" src="https://github.com/user-attachments/assets/461ac5c2-93c0-46dc-b5d4-50e4fb0ddbed">

```
type: custom:apexcharts-card
apex_config:
  chart:
    stacked: true
    animations:
      enabled: true
      easing: easeinout
      animateGradually:
        enabled: true
      dynamicAnimation:
        enabled: true
  xaxis:
    tickAmount: 7
    labels:
      formatter: >
        EVAL:function(value) {
        var datum = new Date(value);
        var doDat = new Date(datum);
        doDat.setTime(datum.getTime() + (3 - ((datum.getDay() + 6) % 7)) * 86400000);
        var kwJahr = doDat.getFullYear();
        var datum1 = new Date(kwJahr, 0, 4);
        var doKW1 = new Date(datum1);
        doKW1.setTime(datum1.getTime() + (3 - ((datum1.getDay() + 6) % 7)) * 86400000);
        var kw = Math.floor((doDat.getTime() - doKW1.getTime()) / 86400000 / 7) + 1;
        return kw - 6;
        }
  yaxis:
    decimalsInFloat: 0
  legend:
    show: false
  dataLabels:
    offsetY: -7
    background:
      enabled: false
    style:
      colors:
        - "#000000"
      fontSize:
        - 8pt
      fontWeight:
        - Regular
graph_span: 56d
span:
  start: isoWeek
show:
  last_updated: false
header:
  show: true
  show_states: false
  colorize_states: true
  title: Gas m³ | Woche
series:
  - entity: sensor.vicare_heating_gas_consumption_this_week
    name: Heizung
    type: column
    unit: " m³"
    color: "#DA6F41"
    offset: "-7w"
    group_by:
      func: last
      duration: 7d
    show:
      datalabels: false
      legend_value: false
  - entity: sensor.vicare_hot_water_gas_consumption_this_week
    name: Warmwasser
    type: column
    unit: " m³"
    color: "#E9984B"
    offset: "-7w"
    group_by:
      func: last
      duration: 7d
    show:
      datalabels: total
      legend_value: false

```

**Reply by Digitalfm** [reply]
Hi, thank you. I figured it out in the end using the 'showForSingleSeries' command. 

I use the below which replicates the legend on the original graphs:

  legend:
    show: true
    showForSingleSeries: true
    position: top
    offsetY: 15
    horizontalAlign: center
    floating: false
    fontSize: 14px
    markers:
      size: 7
      strokeWidth: 1
      offsetY: -0.5
      offsetX: -3
